game/src/Systems/HealthBarSystem.hpp

```diff
@@ -7,64 +7,117 @@
 class HealthBarSystem : public System
 {
 private:
+	const static int MAX_ENTITIES = 30;
+
 	int backgroundID;
 	int healthBarID;
 	Scene* scene;
 	UIRenderer* uiRenderer;
+	std::vector<glm::vec4> backgroundRects;
 public:
 	HealthBarSystem(int backgroundID, int healthBarID, Scene* scene, UIRenderer* uiRenderer)
 		: backgroundID(backgroundID), healthBarID(healthBarID), scene(scene), uiRenderer(uiRenderer)
-	{ }
+	{
+		this->backgroundRects.resize(MAX_ENTITIES);
+	}
 
 	virtual ~HealthBarSystem()
 	{ }
 
 	bool update(entt::registry& reg, float deltaTime) final
 	{
+		if (!this->scene->entityValid(this->scene->getMainCameraID())) { return false; }
+
 		auto tankView = reg.view<Transform, TankComponent>(entt::exclude<Inactive>);
 		auto lichView = reg.view<Transform, LichComponent>(entt::exclude<Inactive>);
 		auto swarmView = reg.view<Transform, SwarmComponent>(entt::exclude<Inactive>);
-		if (!this->scene->entityValid(this->scene->getMainCameraID())) { return false; }
+		int counter = 0;
+		Transform& camTransform = this->scene->getComponent<Transform>(this->scene->getMainCameraID());
 
 		uiRenderer->setTexture(backgroundID);
 		auto tankBackground = [&](Transform& transform, TankComponent& tank)
 		{
-			uiRenderer->renderTexture(transform.position + glm::vec3(0.0f, 36.0f, 0.0f), glm::vec2(2000.0f, 150.0f));
+			float dotDist = glm::dot(transform.position - camTransform.position, camTransform.forward());
+			float alpha = dotDist <= 100.0f ? 1.0f : std::max(125.0f - dotDist, 0.0f) / 25.0f;
```

Comment:
std::clamp() can be used instead of ternary operator. And also, could constants be used instead of hardcoded 125 and 25?